scripts/alert_emails/create_lists_to_email.ts

```diff
@@ -1,69 +1,95 @@
-//@ts-ignore createsend has no types and throws an error
+import _ from 'lodash';
 import fs from 'fs-extra';
 import path from 'path';
-import { Alert } from './interfaces';
 import { getFirestore } from './firestore';
 
+const ERROR_CODE_ALREADY_EXISTS = 6;
+
+function exitWithUsage(): never {
+  console.log('Usage: yarn create-lists-to-email alerts.json snapshotId');
+  process.exit(1);
+}
+
+const readAlertsFile = (alertsFileName: string) => {
+  const filePath = path.join(__dirname, alertsFileName);
+  const rawData = fs.readFileSync(filePath, 'utf8');
+  return JSON.parse(rawData);
+};
+
+// Taken from https://ui.dev/validate-email-address-javascript/
+const EMAIL_REGEX = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+const isValidEmail = (email: string): boolean => EMAIL_REGEX.test(email);
+
 /**
- * Takes a list of fips to alert on and a list of users who want to get alert
- * and generates a set of users to email in firestore like:
- *     - snapshots/<id>/locations/<fips>/emails/<email> => {sent: null }
+ * Takes the alerts file generated by `generate_daily_alerts.ts` and updates the
+ * list of emails that we need to send for each location. The steps are:
+ *
+ * 1. Read the locations that changed threat level from the alerts file.
+ * 2. Fetch the emails that are subscribed to the locations that changed threat level.
+ * 3. Creates a list of emails for each location and stores them in Firestore.
  *
- * Takes in a file with {[fips: string]: {...data}} and generates documents
- * for each user to be emailed.
+ * Run the script via yarn:
  *
- * You can run via `yarn create-lists-to-email fipsToAlertFilename currentSnapshot`
+ *   $ yarn create-lists-to-email alerts.json snapshotId
  */
-(async () => {
-  const outputFolder = path.join(__dirname);
-  await fs.ensureDir(outputFolder);
+const updateSnapshotEmails = async (
+  alertsFileName: string,
+  snapshotId: string,
+) => {
```

Comment:
nit: Is there a reason to declare functions like this (`const foo = ...` instead of `async function foo(...)` ?